snapshot-labs/snapshot
# feat: Space settings to allow voting with alias

### Issues
#3265 
#3852 

### Changes 
1. Added an option to vote with aliases. To enable aliases, go to Settings -> Voting -> Allow actions with session keys
<img width="777" alt="Screenshot 2023-06-07 at 20 27 14" src="https://github.com/snapshot-labs/snapshot/assets/7693369/11e2c468-79ee-4cc5-9603-691e92a01ae6">


### How to test
1. Go to the active proposal
2. Create a proposal, delete it (you will be asked to sign your request)
3. Vote in the usual way (to check if it still works)
4. go to Settings -> Voting. And enable `Allow actions with session keys`. Save settings
5. Go back to an active proposal (the previous one or a new active one)
6. Try to vote (the alias voting). In this case, you will be proposed to sign an alias (only once). But next votings will go smoothly without any signing messages
7. Try to create/delete proposal. It should happen without signing message with wallet
8. Go back to Settings -> Voting -> disable `Allow actions with session keys` -> Save settings
9. Go back to an active proposal and try to vote again. This time it will ask you to sign a message for each voting
10. Create/delete a proposal. It should require to sign request


### To-Do


### Self-review checklist
